docs/docs/genai/prompt-registry/optimize-prompts.mdx

```diff
@@ -169,14 +170,120 @@ def predict(question: str, prompt_uri: str) -> str:
 
 You can customize the optimization process using `OptimizerConfig`, which includes the following parameters:
 
+- **algorithm**: The optimization algorithm to use. Can be a string (e.g., "DSPy/MIPROv2") or a custom optimizer class. Default: "DSPy/MIPROv2"
 - **num_instruction_candidates**: The number of candidate instructions to try. Default: 6
 - **max_few_show_examples**: The maximum number of examples to show in few-shot demonstrations. Default: 6
 - **optimizer_llm**: The LLM to use for optimization. Default: None (uses target LLM)
+- **extract_instructions**: Whether to extract instructions from the initial prompt template. Default: True
 - **verbose**: Whether to show optimizer logs during optimization. Default: False
 - **autolog**: Whether to log the optimization parameters, datasets and metrics. If set to True, a MLflow run is automatically created to store them. Default: False
 
 See <APILink fn="mlflow.genai.OptimizerConfig" /> for more details.
 
+## Custom Optimizers
+
+MLflow supports creating custom prompt optimization algorithms by extending the base optimizer classes. This allows you to implement domain-specific optimization strategies or integrate with other optimization libraries.
+
+### BasePromptOptimizer
+
+For completely custom optimization logic, extend the <APILink fn="mlflow.genai.optimize.BasePromptOptimizer" >BasePromptOptimizer</APILink> class:
```

Comment:
Do we need `completely`?

Reply:
I thought it's clear to say "completely" to compare BasePromptOptimizer and DSPyPromptOptimizer